src/config_reader.cpp

```diff
@@ -224,14 +185,14 @@ const ConfigSection *ConfigFile::GetSection(const char *sect_name) const
  * @param key Name of the key (case sensitive).
  * @return The associated value if it exists, else \c nullptr.
  */
-const char *ConfigFile::GetValue(const char *sect_name, const char *key) const
+std::string ConfigFile::GetValue(const std::string &sect_name, const std::string &key) const
 {
 	const ConfigSection *sect = this->GetSection(sect_name);
-	if (sect == nullptr) return nullptr;
+	if (sect == nullptr) return std::string();
 
 	const ConfigItem *item = sect->GetItem(key);
-	if (item == nullptr) return nullptr;
-	return item->value;
+	if (item == nullptr) return std::string();
```

Comment:
We're hiding lack of an item now, as it becomes indistinguishable from an empty value.
Do we use that feature?

Reply:
It's not used, but I added functions `HasValue`/`HasItem` to allow distinguishing between non-existent and explicitly empty values.